### ISSUE TITLE
syzbot WARNING in close_fs_devices (5.9-rc4)

### ISSUE THREAD

**Issue by kdave** [btrfs/fstests#25]
HEAD commit:    e4c26faa Merge tag 'usb-5.9-rc5' of git://git.kernel.org/p.

dashboard link: https://syzkaller.appspot.com/bug?extid=4cfe71a4da060be47502

https://lore.kernel.org/linux-btrfs/0000000000008fbadb05af94b61e@google.com

```
1152 static int close_fs_devices(struct btrfs_fs_devices *fs_devices)
1153 {
1154         struct btrfs_device *device, *tmp;
1155
1156         if (--fs_devices->opened > 0)
1157                 return 0;
1158
1159         mutex_lock(&fs_devices->device_list_mutex);
1160         list_for_each_entry_safe(device, tmp, &fs_devices->devices, dev_list) {
1161                 btrfs_close_one_device(device);
1162         }
1163         mutex_unlock(&fs_devices->device_list_mutex);
1164
1165         WARN_ON(fs_devices->open_devices);
1166         WARN_ON(fs_devices->rw_devices);

1167         fs_devices->opened = 0;
1168         fs_devices->seeding = false;
1169
1170         return 0;
1171 }

It's the 2nd warning, rw_devices > 0 .
```

**Comment by kdave** [comment]
https://lore.kernel.org/linux-btrfs/0000000000008979ac05afd8e916@google.com

**Comment by josefbacik** [comment]
Got it reproduced here, I'll dig into it in the morning.

**Comment by josefbacik** [comment]
https://lore.kernel.org/linux-btrfs/b3a0a629df98bd044a1fd5c4964f381ff6e7aa05.1600777827.git.anand.jain@oracle.com/

**Comment by kdave** [comment]
Reported as fixed by commit cf89af146b7e btrfs: dev-replace: fail mount if we don't have replace item with target device.
